pom.xml

```diff
@@ -212,6 +217,10 @@ THE SOFTWARE.
       <artifactId>jmh-generator-annprocess</artifactId>
       <version>${jmh.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcpkix-jdk18on</artifactId>
```

Comment:
My concern is that this will somehow clash with the BC pulled in via https://plugins.jenkins.io/bouncycastle-api/ in various plugins. Not sure how the dependency tree is resolved—could the JTH version take precedence, thus causing compilation errors or (worse) subtle behavioral differences in `JenkinsRule`-based tests?

Reply:
especially as the upper-bounds rule in plugins does *not* look at test scope deps (which j-t-h will be) but the PCT does and upgrades test scope deps to be aligned.  (the dependency on jupiter-api iun this plugin already causes an issue)

Reply:
(checking effects in https://github.com/jenkinsci/bom/pull/3720)

Reply:
or just use the java api from `sun.security.x509` and add `Add-Exports: java.base/sun.security.x509` to the librarieys manifiest?

Reply:
> `sun.security.x509`

Looks bad, as this could break anytime when upgrading JDK, whereas we can control BC version (and at least it is a proper API). See also https://bugs.openjdk.org/browse/JDK-8058778, closed as won't fix.

Reply:
>whereas we can control BC version

but leak it to plugins test classpaths over what would or would not come from bouncycastle-api and will get automatically updated by PCT to a potentially different version than provided by the bouncycastle-api plugin.

Reply:
An alternate would be to have JTH itself only support running with a specified cert; but move the code to generate certs on demand to `bouncycastle-api-plugin/src/test/java/` with

```xml
<no-test-jar>false</no-test-jar>
```

so that plugins interested in this could share an implementation.